format-common/src/main/java/io/cdap/plugin/format/charset/CharsetTransformingLineRecordReader.java

```diff
@@ -0,0 +1,237 @@
+
+  public CharsetTransformingLineRecordReader(FixedLengthCharset fixedLengthCharset, byte[] recordDelimiter) {
+    this.fixedLengthCharset = fixedLengthCharset;
+    this.recordDelimiterBytes = recordDelimiter;
+  }
+
+  @VisibleForTesting
+  protected CharsetTransformingLineRecordReader(FixedLengthCharset fixedLengthCharset,
+                                                byte[] recordDelimiter,
+                                                SplitLineReader in,
+                                                long start,
+                                                long pos,
+                                                long end,
+                                                int maxLineLength) {
+    this(fixedLengthCharset, recordDelimiter);
+    this.in = in;
+    this.start = start;
+    this.pos = pos;
+    this.end = end;
+    this.maxLineLength = maxLineLength;
+  }
+
+  /**
+   * Initialize method from parent class, simplified for this our use case from the base class.
+   *
+   * @param genericSplit File Split
+   * @param context      Execution context
+   * @throws IOException if the underlying file or decompression operations fail.
+   */
+  public void initialize(InputSplit genericSplit,
+                         TaskAttemptContext context) throws IOException {
+    FileSplit split = (FileSplit) genericSplit;
+    Configuration job = context.getConfiguration();
+    this.maxLineLength = job.getInt(MAX_LINE_LENGTH, Integer.MAX_VALUE);
+    start = split.getStart();
+    end = start + split.getLength();
+    final Path file = split.getPath();
+
+    // open the file and seek to the start of the split
+    final FileSystem fs = file.getFileSystem(job);
+    FSDataInputStream fileIn = fs.open(file);
+
+    SplittableCompressionCodec codec = new FixedLengthCharsetTransformingCodec(fixedLengthCharset);
+    decompressor = codec.createDecompressor();
+
+    final SplitCompressionInputStream cIn =
+      codec.createInputStream(
+        fileIn, decompressor, start, end,
+        SplittableCompressionCodec.READ_MODE.CONTINUOUS);
+    in = new CompressedSplitLineReader(cIn, job,
+                                       this.recordDelimiterBytes);
+    start = cIn.getAdjustedStart();
+    end = cIn.getAdjustedEnd();
+    filePosition = cIn;
+
+    // If this is not the first split, we always throw away first record
+    // because we always (except the last split) read one extra line in
+    // next() method.
+    if (start != 0) {
+      Text t = new Text();
+      start += in.readLine(t, 4096, maxBytesToConsume(start));
+      LOG.info("Discarded line: " + t.toString());
+    }
+    this.pos = start;
+  }
+
+  /**
+   * Returns the maximum of bytes to consume from the input stream
+   * Since the input is compressed, there is no way to accurately determine how many bytes we need to consume.
+   *
+   * @param pos Current file position
+   * @return Number of bytes to consume.
+   */
+  private int maxBytesToConsume(long pos) {
```

Comment:
nit: pos is not used. It may make sense even to remove the method

Reply:
Removed.